.github/workflows/authentication-python-azure.yml

```diff
@@ -0,0 +1,73 @@
+name: Python Authentication Azure
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+    paths: ['authentication/python/azure/**', '.github/workflows/**']
+  pull_request:
+    paths: ['authentication/python/azure/**', '.github/workflows/**']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      relativePath: ./authentication/python/azure
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.4.0
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pipenv
+          pipenv install --dev --skip-lock
+        working-directory: ${{ env.relativePath }}
+      - name: Check code format
+        run: |
+          pipenv run yapf -dr authentication_azure
+        working-directory: ${{ env.relativePath }}
+      - name: Check import order
+        run: |
+          pipenv run isort authentication_azure
+        working-directory: ${{ env.relativePath }}
+      - name: Lint with Pylint
+        run: |
+          pipenv run pylint authentication_azure --disable=W --rcfile=.pylintrc
+        working-directory: ${{ env.relativePath }}
+      - name: Static type checking with Mypy
+        run: |
+          pipenv run mypy authentication_azure
+        working-directory: ${{ env.relativePath }}
+
+  publish:
+    name: Publish distribution to PyPI
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    runs-on: ubuntu-latest
+    needs: [build]
+    env:
+      relativePath: ./authentication/python/azure
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Install flit
+        run: |
+          pip install flit
+        working-directory: ${{ env.relativePath }}
+      - name: Publish the distibution to PyPI
+        run: flit publish
+        working-directory: ${{ env.relativePath }}
+        env:
+          FLIT_INDEX_URL: https://test.pypi.org/legacy/
+          FLIT_USERNAME: __token__
+          FLIT_PASSWORD: ${{ secrets.TEST_PYPI_API_TOKEN }} 
```

Comment:
can you send me the value for this `TEST_PYPI_API_TOKEN` secret privately so I can add it and unblock you for abstractions as well please?